service/lib/y2storage/agama_proposal.rb

```diff
@@ -206,23 +206,28 @@ def drives_with_empty_partition_table(devicegraph)
       devices.select { |d| d.partition_table && d.partitions.empty? }
     end
 
-    # Planned partitions that will hold the given planned devices
+    # Devices for which the mandatory actions must be executed
     #
-    # @return [Array<Planned::Partition>]
-    def partitions_for_clean
-      # The current logic is quite trivial, but this is implemented as a separate method because
-      # some extra logic is expected in the future (eg. considering partitions on pre-existing
-      # RAIDs and more stuff). See the equivalent method at DevicegraphGenerator.
-      planned_devices.partitions
+    # @return [Array<String>] names of partitionable devices
+    def disks_for_clean
+      return drives_names if config.boot_device.nil? || drives_names.include?(config.boot_device)
+
+      drives_names + [config.boot_device]
```

Comment:
NP: maybe more direct with `[drive_names, config.boot_device].flatten.compact.uniq` ?

Reply:
I changed into something similar to your suggestion